validphys2/src/validphys/process_options.py

```diff
@@ -185,21 +183,31 @@ def _hqp_yqq_xq2map(kin_info):
 
 def _hqp_ptq_xq2map(kin_info):
     # Compute x, Q2
-    QMASS2 = TMASS * TMASS
+    QMASS2 = kin_info[_Vars.m_t2]
     Q = np.sqrt(QMASS2 + kin_info[_Vars.pT_t] * kin_info[_Vars.pT_t]) + kin_info[_Vars.pT_t]
     return Q / kin_info[_Vars.sqrts], Q * Q
 
 
 def _hqp_mqq_xq2map(kin_info):
     # Compute x, Q2
-    QQMASS2 = (2 * TMASS) * (2 * TMASS)
+    QQMASS2 = 4 * kin_info[_Vars.m_t2]
     Q = (
         np.sqrt(QQMASS2 + kin_info[_Vars.m_ttBar] * kin_info[_Vars.m_ttBar])
         + kin_info[_Vars.m_ttBar]
     )
     return Q / kin_info[_Vars.sqrts], Q * Q
 
 
+def _inc_xq2map(kin_info):
+    # Compute x, Q2
+    if {"k1", "k2", "k3"} <= kin_info.keys():
+        kin_info[_Vars.m_X2] = kin_info["k2"]
+        kin_info[_Vars.sqrts] = kin_info["k3"]
+
+    mass2 = kin_info.get_one_of(m_W2, m_Z2, m_t2, m_X2)
+    return np.sqrt(mass2) / kin_info[_Vars.sqrts], mass2
```

Comment:
`m_X2` is not a variable.
In this case it would be better to do something like

```
if {"k1 ... 
    m2 = kin_info["k2"]
else:
    m2 = kin_info.get_one_of(...)

and same for sqrts.
```

And then use the formula referring to the two variables you just got directly.